doctr/transforms/modules/pytorch.py

```diff
@@ -29,26 +30,78 @@ def __init__(
         self.preserve_aspect_ratio = preserve_aspect_ratio
         self.symmetric_pad = symmetric_pad
 
-    def forward(self, img: torch.Tensor) -> torch.Tensor:
-        target_ratio = self.size[0] / self.size[1]
-        actual_ratio = img.shape[-2] / img.shape[-1]
-        if not self.preserve_aspect_ratio or (target_ratio == actual_ratio):
+    def forward(
+        self,
+        img: torch.Tensor,
+        target: Optional[np.ndarray] = None,
+    ) -> Union[torch.Tensor, Tuple[torch.Tensor, np.ndarray]]:
+
+        if isinstance(self.size, int):
+            target_ratio = img.shape[-2] / img.shape[-1]
+            actual_ratio = img.shape[-2] / img.shape[-1]
+        elif isinstance(self.size, tuple) or isinstance(self.size, list):
+            target_ratio = self.size[0] / self.size[1]
+            actual_ratio = img.shape[-2] / img.shape[-1]
+        else:
+            raise AssertionError("size should be either a tuple or an int")
+
+        if not self.preserve_aspect_ratio or (
+            target_ratio == actual_ratio and (isinstance(self.size, tuple) or isinstance(self.size, list))
+        ):
+            if target is not None:
+                return super().forward(img), target
             return super().forward(img)
```

Comment:
More of a question: shouldn't we return the target in both cases?
Because the transform will be used in `sample_transforms` most likely, creating a change of arg/return signatures could bring about a few issues later on

Reply:
do you mean returning `None` when there is no target ? 
For the moment if you pass img, target it returns img, target and if you pass img it returns img